### ISSUE TITLE
Latest snapshot forces to use salt database field when using bcrypt

### ISSUE THREAD

**Issue by sNuuFix** [Xephi/AuthMeReloaded#428]
Latest version (5.2) from git master branch forces to use some value in mySQLColumnSalt field when passwordHash is set to BCRYPT, otherwise it fails to save new user because of mysql syntax error, for example: INSERT INTO users(username,password,ip,lastlogin,name,) VALUES ('player','bcrypt_hash_with_salt','0.0.0.0',1452406638176,'Player','salt_from_bcrypt')
with the part "name,)" obviously being invalid.
This error is unintuitive and when resolved, while doing no big harm other than saving useless data in database because bcrypt salts are stored inside the hash itself (irrelevant in size if you don't have millions of users), may cause issues installing for others.

**Comment by ljacqu** [comment]
Hi, would it be possible that your "mySQLColumnEmail" is empty in your config? The handling of the salt column seems fine and is indeed optional. From the SQL you've posted it would seem that the email column name is empty.

**Comment by sNuuFix** [comment]
No, it is defined (mySQLColumnEmail: email), all of column config options are filled and exist in database, except mySQLColumnGroup. It was working with SHA hashes, but this problem occured right after I changed hashes to bcrypt. Right after that I enabled sql query logs and saw that it suddenly tries to insert salt value.

**Comment by ljacqu** [comment]
The only place where we issue such an `INSERT` statement is in MySQL#saveAuth(), and there the email column comes right after the "name" column. Your SQL example `INSERT INTO users(username,password,ip,lastlogin,name,)` shows there is no salt at play (it tries to insert 6 values: username, password, IP, lastlogin, realname and email). With a salt there would be 7 values.

It really seems to me that the email column name is empty somehow. I just tried with the latest dev build using MySQL and the `BCRYPT` hash algorithm and can't reproduce this issue. I would wager that if you switched back to one of the SHA hashes, the error would still remain.

**Comment by sNuuFix** [comment]
The query I copied from log and changed data in it was when salt column config was empty. right after I put some value in it, it generated (username,password,ip,lastlogin,name,salt). I figured out it The thing is I have password registering mode, not email. Emails are set to false at config and doesn't show up in query at all, just as it should be by default. I figured out it was salt only because data it tried to insert contained hash salt part. I edited original comment, originally I put bcrypt_hash_with_salt and salt_from_bcrypt with tags around then and GitHub stripped it out, showing as '', sorry about confusion. I can replicate this again and make a screenshot with exact query it tries to execute and give you full config that generated it if it helps.
